src/utils/utils.js

```diff
@@ -15,3 +19,37 @@ export const setNativeValue = (element, value) => {
     valueSetter.call(element, value);
   }
 };
+
+export const getPxeBootPatch = vm => {
+  const patch = [];
+  const annotations = get(vm, 'metadata.annotations', {});
+  if (annotations[ANNOTATION_FIRST_BOOT]) {
+    if (annotations[ANNOTATION_FIRST_BOOT] === 'true') {
+      patch.push({
+        op: 'replace',
+        path: `/metadata/annotations/${ANNOTATION_FIRST_BOOT}`,
+        value: 'false',
+      });
+    } else {
+      // find bootable disk and change boot order
+      const bootableDiskIndex = getDisks(vm).findIndex(disk => disk.bootOrder === 2);
+      const bootableInterfaceIndex = getInterfaces(vm).findIndex(i => i.bootOrder === 1);
+
+      if (bootableDiskIndex !== -1 && bootableInterfaceIndex !== -1) {
+        patch.push(
+          {
+            op: 'replace',
+            path: `/spec/template/spec/domain/devices/disks/${bootableDiskIndex}/bootOrder`,
+            value: 1,
+          },
+          {
+            op: 'replace',
+            path: `/spec/template/spec/domain/devices/interfaces/${bootableInterfaceIndex}/bootOrder`,
+            value: 2,
```

Comment:
are we sure we want to even include the  vm interface in boot order?

Reply:
I would like to include the boot order. When you try to pxe boot and fail the installation for some reason, next time you run the VM it will try to boot from disk (which wont work because installation failed) and it will fallback to second boot device which will be PXE. Seems like a nice "feature" to me. Its better than failing the boot altogether because if you have boot order defined, the devices without boot order are not considered at all.

Reply:
Is it really the case? I have not tried that, but the usual behavior is that the boot order falls back to the PXE only when the disk is missing.

Reply:
I guess you are right :) Updated PR is now removing the bootOrder prop from NICs.